### ISSUE TITLE
End game brain freeze

### ISSUE THREAD

**Issue by rowlandhill** [ga-wdi-boston/game-project#605]
Sorry this is so long-winded, but I'm having difficulty moving onto the next step.  As of right now, I have a working board that starts with 'X' and alternates between the letters correctly.  It prevents players from using the same square more than once.  I just wrote a function that compares the innerText of each square to check for a winner, and it console.logs when the winner is found.

Where I'm stuck is ending the game after that.  I'm not sure how to ask/Google for functions that shut a board down and take away the ability to click.  Also, I know we will need an array, but I'm a bit worried because I seemingly do not need an array to check for the winner (I'm just cycling through each data attribute and comparing the innerText of the table).  This is what I have for JavaScript to check for the winner:

```
function startGame () {
  document.turn = 'X'
  changeAlert(document.turn + ' always gets to start.  sorry bruh')
}

function changeAlert (txt) {
  document.getElementById('alert').textContent = txt
}

function nextMove (square) {
  if (square.textContent === '') {
    square.textContent = document.turn
    changeTurn()
  } else {
    changeAlert('pick another square, dummy!')
  }
}

function changeTurn () {
  if (document.turn === 'X') {
    document.turn = 'O'
  } else {
    document.turn = 'X'
  }
  if (winnerCheck() === true) {
    console.log('there is a winner')
  } else {
    changeAlert('it\'s ' + document.turn + '\'s turn now')
  }
}
const elementId = document.getElementsByClassName('square')

const winnerCheck = function () {
  if (elementId[0].innerText === elementId[1].innerText && elementId[1].innerText === elementId[2].innerText && elementId[0].innerText)
    console.log(' wins!')
```

This is what I have for html:

```
<div id="container">
      <div id="alert">Turns, winners, etc...</div>
    <table id="gameboard">
      <tr>
        <td class="square" onClick="nextMove(this)" data-square="0"></td>
        <td class="square" onClick="nextMove(this)" data-square="1"></td>
        <td class="square" onClick="nextMove(this)" data-square="2"></td>
      </tr>
      <tr>
        <td class="square" onClick="nextMove(this)" data-square="3"></td>
        <td class="square" onClick="nextMove(this)" data-square="4"></td>
        <td class="square" onClick="nextMove(this)" data-square="5"></td>
      </tr>
      <tr>
        <td class="square" onClick="nextMove(this)" data-square="6"></td>
        <td class="square" onClick="nextMove(this)" data-square="7"></td>
        <td class="square" onClick="nextMove(this)" data-square="8"></td>
      </tr>
    </div>
    </table>
```

**Comment by jordanallain** [comment]
Sounds like you know you need an array, so lets make one! And its important that each cell of the game board corresponds to a specific index in the array, that way we know its an accurate representation. Any thought on how you should approach that?

**Comment by rowlandhill** [comment]
I believe my `const elementId = document.getElementsByClassName('square')` does that for me, because when I type `elementId` in the console of the browser, it returns this array:
`[td.square, td.square, td.square, td.square, td.square, td.square, td.square, td.square, td.square]`

And, I noticed that after I click in the corresponding cell, the innerText updates to whatever filled the cell ('X' or 'O').

That's as much as I think I know, though, and I'm not entirely sure if `elementId` is an array.

**Comment by jordanallain** [comment]
so you have an array of elements, but what you want is an array of the text from each of those elements?

**Comment by rowlandhill** [comment]
I think so, but I'm still not sure why I need an array.  I know I need one, but I don't understand why, because right now, I can just cycle through the winning combinations since I can already check the innerText of the squares.

Right now, my gameboard data attributes correspond to the correct indexes in an array.  I'm still unsure as to how I would connect the array to my gameboard.

I believe the array would have to look like this:

`arrayName = ['','','','','','','','','']`

**Comment by MicFin** [comment]
Sure you can find the winner by checking the `innerText` but then you need an array to represent that to send the array as part of a `game` object back to the API.

**Comment by rowlandhill** [comment]
Ah, ok, that makes sense, thank you.  So to get started with building an array of text, do I need to addEventListener to the click of the data attribute?

**Comment by MicFin** [comment]
I'm not sure I understand the question, can you rephrase and show my some code examples?

**Comment by jordanallain** [comment]
@rowlandhill updates?

**Comment by rowlandhill** [comment]
I'm rewriting my code in jQuery, so I should be good to go :)
I'll just open another ticket if something new comes up.  Thanks!
